sotodlib/site_pipeline/make_nearby_sources_db.py

```diff
@@ -0,0 +1,254 @@
+    obs_list= ctx.obsdb.query(tot_query, query_tags)
+    
+    # output cosampled data for each obs_id
+    for obs in obs_list:
+        obs_id = obs['obs_id']
+        h5_filename = f"nearby_sources_{obs_id.split('_')[1][:4]}.h5"
+        output_filename = os.path.join(output_dir, h5_filename)
+        entry = {'obs:obs_id': obs_id, 'dataset': obs_id}
+        streamed_wafer_slots  = ['ws{}'.format(index) for index, bit in enumerate(obs_id.split('_')[-1]) if bit == '1']
+        try:
+            for ws in wafer_slots:
+                if ws not in streamed_wafer_slots:
+                    for source_name in source_names:
+                        entry[f'{source_name}_{ws}'] = 0                    
+                else:
+                    meta = ctx.get_meta(obs_id, dets={'wafer_slot': ws})
+                    _aman = ctx.get_obs(obs_id, dets=[])
+                    aman = core.AxisManager(meta.dets, _aman.samps)
+                    aman.wrap('timestamps', _aman.timestamps, [(0, 'samps')])
+                    aman.wrap('boresight', _aman.boresight)
+                    
+                    if not 'focal_plane' in list(meta._fields.keys()):
+                        optics.get_focal_plane(aman, 
+                                               x = np.zeros(aman.dets.count),
+                                               y = np.zeros(aman.dets.count),
+                                               wafer_slot=ws,
+                                               config_path=optics_config)
+                    else:
+                        aman.wrap('focal_plane', meta.focal_plane)                        
+                    flag_fp_isnan = (np.isnan(aman.focal_plane.xi)) | \
+                                    (np.isnan(aman.focal_plane.eta)) | \
+                                    (np.isnan(aman.focal_plane.gamma))
+                    aman.restrict('dets', aman.dets.vals[~flag_fp_isnan])
+
+                    nearby_sources = planets.get_nearby_sources(aman, source_list=source_list, distance=distance)
+                    nearby_source_names = []
+                    for _source in nearby_sources:
+                        nearby_source_names.append(_source[0])
+                    for source_name in source_names:
+                        entry[f'{source_name}_{ws}'] = 1 if source_name in nearby_source_names else 0
+            
+            for source_name in source_names:
+                hit_any = False
+                for ws in wafer_slots:
+                    if entry[f'{source_name}_{ws}'] == 1:
+                        hit_any = True
+                entry[f'{source_name}'] = 1 if hit_any else 0
+            
+            man_db.add_entry(entry, 
+                     filename=output_filename, replace=overwrite)
+            logger.info(f"saved: {obs_id}")
+
+        except Exception as e:
+            logger.error(f"Exception '{e}' thrown while processing {obs_id}")
+            continue
```

Comment:
How should errors be processed though?  The standard for site_pipeline scripts, now, is to count these errors, but continue running ... then at the end, if there were any errors, raise an exception.  This helps prefect report that something has failed, while not totally jamming up the processing because of one weird observation.